lib/msf/core/session/provider/single_command_shell.rb

```diff
@@ -59,22 +59,23 @@ def shell_read_until_token(token, wanted_idx=0, timeout=10)
         loop do
           if (tmp = shell_read(-1))
             buf << tmp
-
             # see if we have the wanted idx
-            parts = buf.split("\n#{token}", -1).map { |part| "#{part}\n" }
-
-            if parts.length == parts_needed
-              # cause another prompt to appear (just in case)
-              shell_write("\n")
-              return parts[wanted_idx]
+            unless buf.nil?
+              # normalize the line endings following the token and parse them
+              buf.gsub!("#{token}\n", "#{token}\r\n")
+              parts = buf.split("#{token}\r\n", -1)
+              if parts.length == parts_needed
+                # cause another prompt to appear (just in case)
+                shell_write("\n")
+                return parts[wanted_idx]
+              end
             end
           end
         end
       end
-    rescue
-      # nothing, just continue
```

Comment:
Bit on the fence about removing the rescues here, as we might have modules relying on nil being returned - rather than exploding. Even if exploding is better/expected behavior 🤔

Reply:
Let's test this a bit more, and if it's good - we'll land it, and wait for the screaming 😱